### ISSUE TITLE
TypeError: 'type' object is not subscriptable

### ISSUE THREAD

**Issue by mdantonio** [sbdchd/celery-types#80]
Hello!

I have a problem with the new Task annotations.
I installed version 0.13.0 and started receiving typing errors on Task annotations (from https://github.com/sbdchd/celery-types/pull/75)

```
Missing type parameters for generic type "Task"
```

So added the expected annotations, like:

```
from celery.app.task import Task
from typing import Any

@app.task(bind=True)
def my_task(self: Task[Any, Any]):

```

But now celery (version 5.2.7) fails at runtime with:

```
    self: Task[Any, Any],
          │    │    └ typing.Any
          │    └ typing.Any
          └ <class 'celery.app.task.Task'>

TypeError: 'type' object is not subscriptable
```

What I'm missing? :cry:

**Comment by sbdchd** [comment]
Oh at runtime, yeah I need to update the docs, you'll need to monkey patch Task to accept a generic argument:

```python
 Task.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)  # type: ignore [attr-defined]
```

**Comment by sbdchd** [comment]
Another option is to do something like:


```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    class AnyTask[Any, Any]: ...
else:
    AnyTask = Task

from celery.app.task import Task
from typing import Any

@app.task(bind=True)
def my_task(self: AnyTask): ...
```

**Comment by craiga** [comment]
Would it be possible for this library to provide something like [django_stubs_ext](https://github.com/typeddjango/django-stubs/tree/master/django_stubs_ext) to do this for us?

**Comment by mdantonio** [comment]
> Oh at runtime, yeah I need to update the docs, you'll need to monkey patch Task to accept a generic argument:
> 
> ```python
>  Task.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)  # type: ignore [attr-defined]
> ```

Ok, thank you for the clarification :+1:

**Comment by NixBiks** [comment]
Is it the same if I want to use `from celery.result import AsyncResult`? Is it then safe to do the following in order to allow for generics?

```py
 AsyncResult.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)  # type: ignore [attr-defined]
```

**Comment by sbdchd** [comment]
Yeah monkey patching `__class_getitem__` should always be safe in Celery as they don't use that method at all
